lib/matplotlib/colors.py

```diff
@@ -1992,3 +1994,49 @@ def from_levels_and_colors(levels, colors, extend='neither'):
 
     norm = BoundaryNorm(levels, ncolors=n_data_colors)
     return cmap, norm
+
+
+def brighten_color(color, frac):
+    """A color helper utility to either darken or brighten the given color.
+
+    This function first converts the given color to RGB using
+    `~.colorConverter` and then to
+    HSL using `colorsys.rgb_to_hsl`.  The lightness is modified according
+    to the given fraction, clipped to be between 0 and 1, and then converted
+    back to RGB using `colorsys.hsl_to_rgb`:
+    L = np.clip(L0 * (1.0 + frac), 0., 1.), where L0 is the original
+    lighness of the color in HSL space.
+
+    Parameters
+    ----------
+    color : string, list, hexvalue
+        Any acceptable Matplotlib color value, such as 'red',
+        'slategrey', '#FFEE11', (1,0,0)
+
+    frac : float
+        The amount by which to darken or brighten the color.  Negative
+        darkens, positive brightens.  Lowest valid negative value is -1.
+        Highest positive value is 1/L of the original color, where L is
+        the lightness in HSL space.  Entries outside this range are allowed,
+        but the resulting lightness is clipped between 0. and 1.
+
+    Returns
+    -------
+    color : tuple of floats
+        tuple representing converted rgb values
+
+    """
+
+    frac = float(frac)
+    if not ((frac < np.Inf) and (frac > -np.Inf)):
+        raise ValueError('argument frac must be a finite float')
+
+    rgb = colorConverter.to_rgb(color)
+
+    h, l, s = colorsys.rgb_to_hls(*rgb)
+
+    l *= 1. + float(frac)
```

Comment:
Not clear to me that this is a more useful definition than `l += frac`.

Reply:
I'd argue that most people have no idea what the L value of a color is, but they know they want it 50% lighter.

Reply:
FWIW and based solely on my limited own experience, I would be inclined to agree with @jklymak about the relative vs.  absolute increment scheme.

Reply:
The question here is not whether users understand L, but whether they expect `brighten_color(brighten_color(x, 0,2), 0.2) == brighten_color(x, 0.4)` (as opposed to `0.44`).

Also the `float` is pointless here, as you convert it above

Reply:
Given that the formula is explicitly in the docs, I hope they realize they are incrementing by 20% each time.